jupyterlab/labapp.py

```diff
@@ -831,6 +831,26 @@ def initialize_handlers(self):  # noqa
     def initialize(self, argv=None):
         """Subclass because the ExtensionApp.initialize() method does not take arguments"""
         super().initialize()
+        if self.collaborative:
+            try:
+                import jupyterlab_rtc  # noqa
+            except Exception:
+                self.log.critical("")
+                self.log.critical(
+                    "*********************************************************************************"
+                )
+                self.log.critical(
+                    "To enable real-time collaboration, you must install the extension jupyterlab_rtc."
+                )
+                self.log.critical("You can install it using pip for example:")
+                self.log.critical("")
+                self.log.critical("\tpython -m pip install jupyterlab_rtc")
+                self.log.critical("")
+                self.log.critical(
+                    "*********************************************************************************"
+                )
+                self.log.critical("")
```

Comment:
You should never do that because log messages may be sent to any kind of handlers; some may be for example a log aggregator in which you want one entry per event.

For the same reason you should avoid including any kind of format.

```suggestion
                self.log.critical("""
To enable real-time collaboration, you must install the extension `jupyterlab_rtc`.
You can install it using pip for example:

  python -m pip install jupyterlab_rtc
""")
```